pynvim/util.py

```diff
@@ -40,3 +41,22 @@ def get_client_info(kind, type_, method_spec):
 
 
 VERSION = Version(major=0, minor=3, patch=2, prerelease='')
+
+
+def transform_keyerror(f):
+    """Decorator to transform "Key not found" NvimErrors into KeyErrors."""
+    @functools.wraps(f)
+    def wrapper(*args, **kwargs):
+        # XXX: # not on top, because of circular imports.
+        from .api.nvim import NvimError, NvimKeyError
+
+        try:
+            return f(*args, **kwargs)
+        except Exception as exc:
+            if isinstance(exc, NvimError):
+                if exc.args[0].startswith('Key not found:'):
+                    raise NvimKeyError(exc.args[0])
```

Comment:
Why not just use `KeyError` ? `NvimKeyError` sounds like typical overengineering.

Reply:
There is a test that expects to see `vim.error`.
And it makes sense to have it for backward compatibility I think.

Reply:
hmm, so it is required for vim compat then.

Reply:
Or is it? `vim.current.buffer.vars` and `.option` seem to return `KeyError` only.

Reply:
I've meant compat with previous pynvim versions.
Our test for this: https://github.com/blueyed/python-client/blob/b9f0061c6d862062aa4596478029ddf70472809d/test/test_buffer.py#L156-L161

Reply:
`vim.error` is a vim-compat feature, no need to be compatible with a vim behavior which does not exist. The question is if some plugin is using `NvimError` explicitly to check for missing key which I wouldn't expect (the canonical way if you expect it to be missing is `.get()` ), perhaps checking the matches on github is enough.

Reply:
Could not find usages via https://github.com/search?p=2&q=except+nvimerror+vim+vars+extension%3Apy&type=Code - changed it to use KeyError directly.